apps/aechannel/src/aesc_offchain_update.erl

```diff
@@ -102,8 +103,8 @@ apply_on_trees(Update, Trees0, Round, Reserve) ->
             Trees3 = create_account(ContractPubKey, Trees2),
             Trees4 = add_tokens(ContractPubKey, Deposit, Trees3),
             Call = aect_call:new(OwnerId, Round, ContractId, Round, 0),
-            _Trees = aect_channel_contract:run_new(ContractPubKey, Call, CallData,
-                                                  Round, Trees4);
+            _Trees = aect_channel_contract:run_new(ContractPubKey, Call, CallData, Trees4,
+                                                  OnChainTrees, OnChainEnv);
         {?OP_CALL_CONTRACT, CallerId, ContractId, VmVersion, Amount, CallData,
```

Comment:
As the semantics of this (call contract) off-chain update is part of consensus, shall this not call back into `aect_call_tx` rather than re-implementing most of the steps? See [dual doc comment](https://github.com/aeternity/protocol/pull/215#discussion_r222593537).

(I am ok with any answer, e.g. "no"  or "yes but later", though I am wondering how maintenance of two distinct code paths meant to have same semantics is going to be.)

Strictly speaking this is not part of this ticket/PR so no changes needed for this PR.

Reply:
I guess it would be `yes but later`. Tracked in PT [SC contracts: refactor aect_channel_contract](https://www.pivotaltracker.com/story/show/160992661)